packages/wallet/src/redux/state.ts

```diff
@@ -1,10 +1,85 @@
-import { SharedWalletState, emptyState } from './shared/state';
-import { InitializingState } from './initializing/state';
-import { InitializedState } from './initialized/state';
+import { OutboxState, EMPTY_OUTBOX_STATE } from './outbox/state';
+import { FundingState, EMPTY_FUNDING_STATE } from './fundingState/state';
+import { ChannelState } from './channelState/state';
+import { Properties } from './utils';
 
-export * from './initialized/state';
-export * from './initializing/state';
+export type WalletState = WaitForLogin | WaitForAdjudicator | MetaMaskError | Initialized;
 
-export { SharedWalletState, emptyState };
+// -----------
+// State types
+// -----------
+export const WAIT_FOR_LOGIN = 'INITIALIZING.WAIT_FOR_LOGIN';
+export const METAMASK_ERROR = 'INITIALIZING.METAMASK_ERROR';
+export const WAIT_FOR_ADJUDICATOR = 'INITIALIZING.WAIT_FOR_ADJUDICATOR';
+export const WALLET_INITIALIZED = 'WALLET.INITIALIZED';
 
-export type WalletState = InitializingState | InitializedState;
+// ------
+// States
+// ------
+export interface WaitForLogin {
+  type: typeof WAIT_FOR_LOGIN;
+  channelState: ChannelState;
+  fundingState: FundingState;
+  outboxState: OutboxState;
+}
+
+export interface MetaMaskError {
+  type: typeof METAMASK_ERROR;
+  channelState: ChannelState;
+  fundingState: FundingState;
+  outboxState: OutboxState;
+}
+
+export interface WaitForAdjudicator {
+  type: typeof WAIT_FOR_ADJUDICATOR;
+  channelState: ChannelState;
+  fundingState: FundingState;
+  outboxState: OutboxState;
+  uid: string;
+}
+
+export interface Initialized {
+  type: typeof WALLET_INITIALIZED;
+  channelState: ChannelState;
+  fundingState: FundingState;
+  outboxState: OutboxState;
+  uid: string;
+  networkId: number;
+  adjudicator: string;
+}
+
+// ------------
+// Constructors
+// ------------
+export const emptyState = {
+  outboxState: EMPTY_OUTBOX_STATE,
+  fundingState: EMPTY_FUNDING_STATE,
+  channelState: { initializedChannels: {}, initializingChannels: {} },
+};
+
+export function waitForLogin(): WaitForLogin {
+  return { type: WAIT_FOR_LOGIN, ...emptyState };
+}
+
+export function metaMaskError(params: Properties<MetaMaskError>): MetaMaskError {
+  const { outboxState, fundingState, channelState } = params;
+  return { type: METAMASK_ERROR, outboxState, fundingState, channelState };
+}
+
+export function waitForAdjudicator(params: Properties<WaitForAdjudicator>): WaitForAdjudicator {
+  const { outboxState, fundingState, channelState, uid } = params;
+  return { type: WAIT_FOR_ADJUDICATOR, outboxState, fundingState, channelState, uid };
+}
+
+export function initialized(params: Properties<Initialized>): Initialized {
+  const { outboxState, fundingState, channelState, uid, networkId, adjudicator } = params;
```

Comment:
Minor but is there a reason we can't just do this?:
 ```
return {
    type: WALLET_INITIALIZED,
    ...params
}
```

Reply:
Yeah - `Properties<Initialized>` could also contain extra keys. (I'm not sure if that's actually important here, but it was useful when we were constructing channel states.)